### PR TITLE
worker: Do not use references for async callbacks

### PR DESCRIPTION
**Pull request by jmillan**
If the callback is not executed in the same uv_loop iteration, the access to such values upon executing the callback is undefined.

### CODE REVIEW

**Comment by jmillan** [comment]
TODO: CHANGELOG

File: worker/include/RTC/SenderBandwidthEstimator.hpp [hunk 1]
```diff
@@ -89,7 +89,7 @@ namespace RTC
 	public:
 		void TransportConnected();
 		void TransportDisconnected();
-		void RtpPacketSent(SentInfo& sentInfo);
+		void RtpPacketSent(const SentInfo& sentInfo);
```
Comment (ibc) [inline worker/include/RTC/SenderBandwidthEstimator.hpp:92]: Wow unrelated changes hehe

**Comment by ibc** [comment]
What is this fixing?

**Review by ibc — APPROVED** [review]

File: worker/include/RTC/SenderBandwidthEstimator.hpp (hunk 1 above)
Comment (jmillan) [reply worker/include/RTC/SenderBandwidthEstimator.hpp:92]: Nop :-)

Screenshot 2023-12-20 at 15 30 07

**Comment by jmillan** [comment]
> What is this fixing?

If the packet is not sent in the current loop iteration, ie: if `uv_udp_try_send()` cannot send the packet immediately and we need to use `uv_udp_send()` which will send the packet when possible (not in this iteration) and notify us when that is done,  we will run the send callback in the future. In such future, the variables passed by reference to the callback will NOT exist, because they were in the **stack**, back in such previous iteration when `uv_udp_send()` was called.

Accessing references to stack variables that do not exist anymore has undefined behaviour, so we may be probably calling the transport congestion `PacketSent()` methods with garbage.

Also, this reminded me that we were having a crash with `jemalloc` at work, and this may well be related.

File: worker/include/RTC/SenderBandwidthEstimator.hpp (hunk 1 above)
Comment (jmillan) [reply worker/include/RTC/SenderBandwidthEstimator.hpp:92]: This is was a needed change as we are not creating the `onSendCallback()` out of references.

**Comment by ibc** [comment]
I see. Good. Approved, like my txuleton. Look:

image

**Comment by jmillan** [comment]
> I see. Good. Approved, like my txuleton

0xdeadbeef

**Comment by ibc** [comment]
Plus good dessert and 3 wines and patxaran
